### PR TITLE
[WFCORE-6815][Community] Added reload-enhanced operation to reload a domain to a certain stability

### PR DESCRIPTION
**Pull request by yersan**
Jira issue: https://issues.redhat.com/browse/WFCORE-6815

WildFly Proposal: https://github.com/wildfly/wildfly-proposals/pull/570

### CODE REVIEW

**Comment by yersan** [comment]
@kabir This PR is based on your previous work for standalone, although it has some particularities to adapt it on Domain Mode, could you take a look and provide your feedback? thanks

File: host-controller/src/main/java/org/jboss/as/host/controller/HostControllerEnvironment.java
Comment (yersan) [inline host-controller/src/main/java/org/jboss/as/host/controller/HostControllerEnvironment.java:940]: Notice we also have https://issues.redhat.com/browse/WFCORE-6760 to move common functionality to ProcessEnvironment, that can be done later

**Review by kabir — COMMENTED** [review]
I think it looks good

File: host-controller/src/main/java/org/jboss/as/domain/controller/operations/HostProcessReloadHandler.java
Comment (yersan) [reply host-controller/src/main/java/org/jboss/as/domain/controller/operations/HostProcessReloadHandler.java:87]: Thanks for reviewing @kabir 

What do you mean about having the same format? 

I understand that what you suggested is to remove the overloaded DeferredParametersOperationDefinitionBuilder constructor, so both `getEnhancedDefinition` and `getDefinition` use the same number of arguments to improve the code readability and make it more clear, correct?

Comment (yersan) [reply host-controller/src/main/java/org/jboss/as/domain/controller/operations/HostProcessReloadHandler.java:87]: So far, just updated it to assert that closing the snapshot indeed reloaded the host controller to the original stability level

**Comment by yersan** [comment]
@kabir Friendly remainder, this one is waiting for your approval to get it merged, please take a look, thanks!

**Review by kabir — APPROVED** [review]

**Comment by yersan** [comment]
Thanks @kabir  @jamezp for the reviews here and in the analysis doc

**Pull request merged**
